client/lib/i18n-utils/switch-locale.js

```diff
@@ -11,14 +11,16 @@ import { noop } from 'lodash';
 /**
  * Internal dependencies
  */
-import { isDefaultLocale, getLanguage } from './utils';
+import { isDefaultLocale, getLanguage, getLangRevision } from './utils';
 
 const debug = debugFactory( 'calypso:i18n' );
 
 function languageFileUrl( localeSlug ) {
 	const protocol = typeof window === 'undefined' ? 'https://' : '//'; // use a protocol-relative path in the browser
+	const revision = getLangRevision( localeSlug );
+	const fileUrl = `${ protocol }widgets.wp.com/languages/calypso/${ localeSlug }.json`;
 
-	return `${ protocol }widgets.wp.com/languages/calypso/${ localeSlug }.json`;
+	return revision ? fileUrl + `?v=${ revision }` : fileUrl;
```

Comment:
Can we re-use this code for https://github.com/Automattic/wp-calypso/pull/26947/files#r213830884? Maybe encapsulate in a helper function?

Reply:
(Well, `languageFileUrl` can probably be that helper function :grin: )